apis/v1beta1/types.go

```diff
@@ -220,8 +220,31 @@ type VSphereDisk struct {
 	// SizeGiB is the size of the disk in GiB.
 	// +kubebuilder:validation:Required
 	SizeGiB int32 `json:"sizeGiB"`
+	// provisioningType specifies the provisioning type to be used for creating this disk for VirtualMachine.
+	// This does not apply to PersistentVolumeClaim volumes that are created and managed externally.
+	// If not set, the setting will be provided by the default storage policy.
+	// +optional
+	ProvisioningType ProvisioningType `json:"provisioningType,omitempty"`
```

Comment:
```suggestion
	ProvisioningMode ProvisioningMode `json:"provisioningMode,omitempty"`
```

Let's go with mode, that's whats used in supervisor.